### DISCUSSION TITLE
[SDK] use zksync-ethers with typechain

### DISCUSSION THREAD

**Discussion by nooma-42** [zkSync-Community-Hub/zksync-developers#348] [SDKs] [answered]
### Environment

Testnet

### Acknowledgement

- [ ] I understand that issues are expected with the Go SDK as they are being updated.

### Issue Description

How to use zksync-ethers with typechain?

typechain generate `__factory` files and export in `index.ts`, and factory extend ContractFactory in ethers, like this:
```typescript
export class TimelockController__factory extends ContractFactory {
```
however, I should use ContractFactory from zksync-ethers in the second argument of `handleDeploy`. Clearly mismatch the type. like below:
```typescript
return this.multiProvider.handleDeploy(
      chain,
      new TimelockController__factory(),
      // delay, [proposers], [executors], admin
      [
        timelockConfig.delay,
        [timelockConfig.roles.proposer],
        [timelockConfig.roles.executor],
        ethers.constants.AddressZero,
      ],
    );
```

Should I use a wrapper to make the typechain generated factory compatible with zksync-ethers or there're other standardized way to handle such cases?

### Expected Behavior

N/A

### Code Example

N/A

**Comment by MexicanAce** [comment, marked as answer]
Hi @NOOMA-42 👋 !

Thank you for the question, but this question is best suited for the Typechain GitHub repo : https://github.com/dethcrypto/TypeChain/issues

Thank you!
